### ISSUE TITLE
Use msys2 on windows

### ISSUE THREAD

**Issue by Stebalien** [protocol/.github#192]
We currently use git bash as the default bash shell. Unfortunately, the build environment is a bit lacking. At the moment, building with openssl support on windows is tricky to say the least. With msys2, we can install openssl and pkgconfig with `pacman`.

Note: msys2 is already installed by default, but it's easier to configure with https://github.com/msys2/setup-msys2.

**Comment by marten-seemann** [comment]
How would we do that? Replace `shell: bash` with `shell: msys2` in all `run` steps if `matrix.os == 'windows'`?

**Comment by Stebalien** [comment]
I think we can do this on a per-job basis:

```yaml
strategy:
  matrix:
    os: [ ... ]
    include: 
      - os: windows-latest
         shell: ...
      - os: ...
defaults:
  run:
    shell: ${{ matrix.shell }}
```

But there may be a more elegant approach?

**Comment by Stebalien** [comment]
We can hack around this in go-openssl for now.

**Comment by galargh** [comment]
I've merged https://github.com/protocol/.github/pull/232 into the `next` branch which enables `msys2` on windows machines.

**Comment by marten-seemann** [comment]
Fixed by #232.
